### PR TITLE
Worksheet B

### PR DESCRIPTION
**Pull request by Phil-Sparkes**

### CODE REVIEW

**Comment by Phil-Sparkes** [comment]
Added worksheet c

File: worksheet_B/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+```
+while IncorrectGuesses < 4                                      # loops until the user has 4 inccorect guesses
```
Comment (edpowley) [inline worksheet_B/README.md:2]: Pseudocode looks good 👍 

Minor nitpicks: you should specify that IncorrectGuesses is initialised to 0 and SecretWord is initialised to a randomly selected word.

File: worksheet_B/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        IncorrectGuesses = IncorrectGuessses + 1                # user did not guess correctly - adds one to the incorrect counter
+        print "Incorrect guesses " + IncorrectGuesses + "/4"    # shows the user how many incorrect guesses they have
+    end if
+
+end while
+
+print "game over"                                               # if the user has 4 incorrect guesses then "game over" is displayed
+```
+
+![flowchart](Flowchart.png)
```
Comment (edpowley) [inline worksheet_B/README.md:30]: Flowchart looks good 👍

File: worksheet_C/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+(a) Checks if any items on the list are repeated
```
Comment (edpowley) [inline worksheet_C/README.md:1]: ✅

File: worksheet_C/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+(b) It's taking a value from the list O(n) and checking it against every other value of the list O(n^2)
```
Comment (edpowley) [inline worksheet_C/README.md:3]: ✅

File: worksheet_C/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+(a) Checks if any items on the list are repeated
+
+(b) It's taking a value from the list O(n) and checking it against every other value of the list O(n^2) 
+
+(c) It still checks every item on the list against every other item
```
Comment (edpowley) [inline worksheet_C/README.md:5]: True, but needs more explanation (i.e. explain that the original algorithm checks each pair in both possible orders, whereas the new one checks each pair only once)

File: worksheet_C/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+(d) It’s not going over results that it has already checked
```
Comment (edpowley) [inline worksheet_C/README.md:7]: Again, needs more explanation -- you have the right idea, but take care to explain your answers in sufficient detail.

File: worksheet_C/README.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+(e) Yes it’s still quadratic ((n^2)-n)/2
```
Comment (edpowley) [inline worksheet_C/README.md:9]: ✅ but explain why this is quadratic, i.e. n^2 dominates n, ignore the factor of 1/2

File: worksheet_C/README.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+
+(f) O(n log n) http://bugs.java.com/bugdatabase/view_bug.do?bug_id=6804124
```
Comment (edpowley) [inline worksheet_C/README.md:11]: ✅ but a better source would be the Timsort documentation linked from that page, or even better the official Python documentation wiki.

File: worksheet_C/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+(g) O(n) The term that is largest when n is large. O(n log n) is combined with O(n) it becomes O(n log n) + O(n)
```
Comment (edpowley) [inline worksheet_C/README.md:13]: Right reasoning but wrong answer -- n log n > n for large n, so the answer is O(n log n).

File: worksheet_C/README.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+
+(h) The Second - Linear is faster than quadratic given a large input.
```
Comment (edpowley) [inline worksheet_C/README.md:15]: ✅

File: worksheet_C/README.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+
+(i) Lack of temporary storage space.
```
Comment (edpowley) [inline worksheet_C/README.md:17]: ✅ but needs more explanation (i.e. explain that sorting is likely to require temporary storage, possibly backing this up with a reference)

**Pull request closed without merging**